src/Console/InfectionContainer.php

```diff
@@ -204,15 +205,24 @@ public static function create(): self
             VersionParser::class => static function (): VersionParser {
                 return new VersionParser();
             },
-            'lexer' => static function (): Lexer {
+            Lexer::class => static function (): Lexer {
                 return new Lexer\Emulative([
                     'usedAttributes' => [
                         'comments', 'startLine', 'endLine', 'startTokenPos', 'endTokenPos', 'startFilePos', 'endFilePos',
                     ],
                 ]);
             },
-            'parser' => static function (self $container): Parser {
-                return (new ParserFactory())->create(ParserFactory::PREFER_PHP7, $container['lexer']);
+            Parser::class => static function (self $container): Parser {
+                /** @var Lexer $lexer */
+                $lexer = $container[Lexer::class];
+
+                return (new ParserFactory())->create(ParserFactory::PREFER_PHP7, $lexer);
+            },
+            FileParser::class => static function (self $container): FileParser {
+                /** @var Parser $phpParser */
+                $phpParser = $container[Parser::class];
```

Comment:
I wouldn't mind using the values directly, instead of assigning a temporary variable, but thats not blocking for me

Reply:
It's ultimately what I would like to do but I think it's better to do so once we introduce getters in the container for the IDE to be able to infer the type. Otherwise right now if you do "Find by usage" the IDE will not find it and SA have trouble with it as well